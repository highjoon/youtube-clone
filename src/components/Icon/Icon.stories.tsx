/* eslint-disable react/no-unknown-property */
import { css } from '@emotion/react';

import Icon, { iconTypes } from './Icon';

export default {
  component: Icon,
  title: 'components/Icon',
};

export const Icons = () => {
  return (
    <ul css={iconListStyle}>
      {iconTypes.map(icon => (
        <li key={icon}>
          <Icon icon={icon} size={'50px'} />
          {icon}
        </li>
      ))}
    </ul>
  );
};

const iconListStyle = css`
  display: flex;
  flex-wrap: wrap;
  list-style: none;

  li {
    box-sizing: border-box;
    width: 25%;
    padding: 1rem;
    display: flex;
    align-items: center;
    svg {
      margin-right: 1rem;
    }
  }
`;
