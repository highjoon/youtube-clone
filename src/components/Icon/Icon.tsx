/* eslint-disable @typescript-eslint/no-explicit-any */

import React from 'react';

import * as icons from '@/assets/icons';

export type IconType = keyof typeof icons;
export const iconTypes: IconType[] = Object.keys(icons) as any[];

export interface IconProps {
  icon: IconType;
  color?: string;
  size?: string | number;
  className?: string;
}

const Icon = ({ icon, color, size = 24, className }: IconProps) => {
  // eslint-disable-next-line import/namespace
  const Icon = icons[icon];

  return (
    <Icon
      data-testid={'icon'}
      className={className}
      css={{
        fill: color || 'currentcolor',
        width: size,
        height: 'auto',
      }}
    />
  );
};

export default Icon;
