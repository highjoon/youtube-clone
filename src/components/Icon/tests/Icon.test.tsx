import { render } from '@testing-library/react';

import Icon from '../Icon';

describe('Icon', () => {
  const IconComponent = <Icon icon="Home" />;
  it('정상적으로 렌더링됩니다.', () => {
    const { asFragment } = render(IconComponent);

    expect(asFragment()).toMatchSnapshot();
  });
});
