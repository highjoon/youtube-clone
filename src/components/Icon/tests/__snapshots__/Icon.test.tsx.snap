// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon 정상적으로 렌더링됩니다. 1`] = `
<DocumentFragment>
  <icon-mock
    classname="css-uveo7x-Icon"
    data-testid="icon"
  />
</DocumentFragment>
`;
